### ISSUE TITLE
Possible race condition in mailbox_watcher.rb quit method - undefined method `join' for nil:NilClass

### ISSUE THREAD

**Issue by galet** [tpitale/mail_room#77]
In our GitLab deployment we are running into the following error:

```
2016-07-26_08:31:54.08717 /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/mailbox_watcher.rb:50:in `quit': undefined method `join' for nil:NilClass (NoMethodError)
2016-07-26_08:31:54.08721       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/coordinator.rb:30:in `each'
2016-07-26_08:31:54.08721       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/coordinator.rb:30:in `quit'
2016-07-26_08:31:54.08722       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/coordinator.rb:25:in `ensure in run'
2016-07-26_08:31:54.08722       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/coordinator.rb:25:in `run'
2016-07-26_08:31:54.08722       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/lib/mail_room/cli.rb:52:in `start'
2016-07-26_08:31:54.08723       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/mail_room-0.8.0/bin/mail_room:5:in `<top (required)>'
2016-07-26_08:31:54.08723       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/bin/mail_room:23:in `load'
2016-07-26_08:31:54.08724       from /opt/gitlab/embedded/service/gem/ruby/2.1.0/bin/mail_room:23:in `<main>'
```

It seems to be caused by some kind of race condition.

Original issue: https://gitlab.com/gitlab-org/gitlab-ce/issues/20273

**Comment by marceldegraaf** [comment]
@tpitale this is breaking Gitlab CE currently when using IMAP to fetch email. Can you push a release so it can be included in the next Gitlab release? Thanks!

**Comment by tpitale** [comment]
Just released v0.8.1 https://rubygems.org/gems/mail_room/versions/0.8.1
